src/qibolab/platforms/abstract.py

```diff
@@ -16,6 +17,85 @@
+
+    Attributes:
+        instruments (list): List of the three instrument names (str) controlling this qubit.
+        channels (list): List of the three channels (int) associated to the qubit.
+        ro_channel, qd_channel, qf_channel (int): Explicit references to the channels.
+        ro_port, qd_port, qf_port: Explicit references to the ports.
+    """
+
+    def __init__(self, index, settings):
+        self.index = index
+        self.channels = settings["qubit_channel_map"][index]
+        self.ro_channel, self.qd_channel, self.qf_channel = self.channels
+
+        # Generate qubit_instrument_map from qubit_channel_map and the instruments' channel_port_maps
+        self.instruments = [None, None, None]
+        for name, value in settings.get("instruments", {}).items():
+            instrument_settings = value.get("settings")
+            channel_port_map = instrument_settings.get("channel_port_map", [])
+            s4g_modules = instrument_settings.get("s4g_modules", [])
+            for channel in itertools.chain(channel_port_map, s4g_modules):
+                if channel in self.channels:
+                    self.instruments[self.channels.index(channel)] = name
+
+        self.ro_port, self.qd_port, self.qf_port = None, None, None
+
+        self.native_one_qubit = settings["native_gates"]["single_qubit"][index]
+        # TODO: Think how to implement two qubit gates
+
+    def set_ports(self, instruments):
+        """Set instrument ports (ro_port, qd_port, qf_port) associated to this qubit.
+
+        Args:
+            instruments (dict): Dictionary mapping instrument name to instrument
+                objects, created in :meth:`AbstractPlatform.__init__`.
+        """
+        if self.instruments[0] is not None:
+            qrm = instruments[self.instruments[0]]
+            self.ro_port = qrm.ports[qrm.channel_port_map[self.channels[0]]]
+        if self.instruments[1] is not None:
+            qcm = instruments[self.instruments[1]]
+            self.qd_port = qcm.ports[qcm.channel_port_map[self.channels[1]]]
+        if self.instruments[2] is not None:
+            qbm = instruments[self.instruments[2]]
+            self.qf_port = qbm.dacs[self.channels[2]]
+
+    def get_native_gate(self, name, start, relative_phase):
+        """Get one-qubit native gate acting on this qubit.
+
+        This maps the gate to a single pulse and works for the native gates
+        provided in the runcard.
+
+        Args:
+            name (str): Name of the native gate (should agree with the runcard).
+            start (float): Start time for the corresponding pulse.
+            relative_phase (float): Relative phase of the corresponding pulse.
+
+        Returns
+            kwargs (dict): Dictionary containing all the arguments required
+                for constructing the single pulse that implements the given gate.
+        """
+        kwargs = dict(self.native_one_qubit.get(name))
+        kwargs.pop("phase", None)
```

Comment:
Hi, this should not be necessary:
```python
kwargs.pop("phase", None)
```
neither `phase` nor `start` should be parameters of the native gates in the runcard (I'm not sure who added them).

`phase` should be replaced with `relative_phase` or removed altogether and then let the platform set the pulse appropriate relative_phase depending on the type of gate (RX -> relative_phase=0) (RY -> relative_phase=pi/2).
`start`: it does not make sense to set start value to a gate in the runcard.